test/unit/mdc-chips/mdc-chip-set.foundation.test.js

```diff
@@ -34,118 +34,75 @@ test('exports cssClasses', () => {
 
 test('defaultAdapter returns a complete adapter implementation', () => {
   verifyDefaultAdapter(MDCChipSetFoundation, [
-    'hasClass', 'removeChip',
+    'hasClass', 'removeChip', 'setSelected',
   ]);
 });
 
 const setupTest = () => {
   const mockAdapter = td.object(MDCChipSetFoundation.defaultAdapter);
   const foundation = new MDCChipSetFoundation(mockAdapter);
-  const chipA = td.object({
-    foundation: {
-      isSelected: () => {},
-      setSelected: () => {},
-    },
-  });
-  const chipB = td.object({
-    foundation: {
-      isSelected: () => {},
-      setSelected: () => {},
-    },
-  });
-  return {foundation, mockAdapter, chipA, chipB};
+  return {foundation, mockAdapter};
 };
 
-test('in choice chips, #handleChipInteraction selects chip if no chips are selected', () => {
-  const {foundation, mockAdapter, chipA} = setupTest();
+test('in choice chips, #select selects chip if no chips are selected', () => {
+  const {foundation, mockAdapter} = setupTest();
   td.when(mockAdapter.hasClass(cssClasses.CHOICE)).thenReturn(true);
-  td.when(chipA.foundation.isSelected()).thenReturn(false);
-  assert.equal(foundation.selectedChips_.length, 0);
+  assert.equal(foundation.getSelectedChipIds().length, 0);
 
-  foundation.handleChipInteraction({
-    detail: {
-      chip: chipA,
-    },
-  });
-  td.verify(chipA.foundation.setSelected(true));
-  assert.equal(foundation.selectedChips_.length, 1);
+  foundation.select('chipA');
+  td.verify(mockAdapter.setSelected('chipA', true));
+  assert.equal(foundation.getSelectedChipIds().length, 1);
 });
 
-test('in choice chips, #handleChipInteraction deselects chip if another chip is selected', () => {
-  const {foundation, mockAdapter, chipA, chipB} = setupTest();
+test('in choice chips, #select deselects chip if another chip is selected', () => {
```

Comment:
Add another test for `in choice chips, #select does not affect selected state if the same chip is selected` and test that `getSelectedChipIds().length` is 1